android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -562,22 +562,30 @@ private void doMenuAction(int menuItemId) {
 
     @Override
     public void onBackPressed() {
+        Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.content_frame);
+        boolean mapFragment = SharedPreferencesUtil.getBoolean(ConstantStrings.IS_MAP_FRAGMENT_FROM_MAIN_ACTIVITY, false);
```

Comment:
I meant this shouldn't be needed. This can be done in memory

Reply:
@iamareebjamal  can you please tell me what should I use to communicate this message from fragment to activity?

Reply:
Activity can implement an interface whose instance will be passed in Fragment. Like callbacks in javascript